### ISSUE TITLE
calico tunnels can not ping each other 

### ISSUE THREAD

**Issue by ehsanrocket** [projectcalico/calico#3142]
hi all 
I have set up a kubernetes cluster with a master node and two worker node. and calico for the network.
tunnels can not ping each others.
Kubernetes Version: 1.17
Calico Version: v3.11.1

Master IP : 1.1.1.1/22
Node1 IP: 2.2.2.2/22
Node2 IP: 3.3.3.3/23
IP's are these numbers, but are valid IP's.

`watch -n 0 kubectl get pods --all-namespaces ` output is :
```
on master : NAMESPACE              NAME                                 READY   STATUS    RESTARTS   AGE
default                nginx-86c57db685-6754h                           1/1     Running   0          178m
default                tvts-c667fd89b-cwfx2                             1/1     Running   0          175m
default                tvts-c667fd89b-wgbnq                             1/1     Running   0          176m
kube-system            calico-kube-controllers-5b644bc49c-zwkk2         1/1     Running   0          31m
kube-system            calico-node-n7s68                                1/1     Running   0          31m
kube-system            calico-node-w4p52                                1/1     Running   0          31m
kube-system            calico-node-zdghr                                1/1     Running   0          31m
kube-system            coredns-6955765f44-9d4bc                         1/1     Running   0          3h4m
kube-system            coredns-6955765f44-x86pf                         1/1     Running   0          3h4m
kube-system            etcd-srvarvan-kube                               1/1     Running   0          3h4m
kube-system            kube-apiserver-srvarvan-kube                     1/1     Running   0          3h4m
kube-system            kube-controller-manager-srvarvan-kube            1/1     Running   0          3h4m
kube-system            kube-proxy-9sbvb                                 1/1     Running   0          3h4m
kube-system            kube-proxy-nc86c                                 1/1     Running   0          179m
kube-system            kube-proxy-xws62                                 1/1     Running   0          178m
kube-system            kube-scheduler-srvarvan-kube                     1/1     Running   0          3h4m
kubernetes-dashboard   dashboard-metrics-scraper-76585494d8-2mcn8       1/1     Running   0          3h1m
kubernetes-dashboard   kubernetes-dashboard-f9bd45cd6-t4j6d             1/1     Running   0          3h1m
```
on node Master (1.1.1.1)
```
sudo calicoctl node status
+-----------------+-------------------+-------+----------+-------------+
|  PEER ADDRESS   |     PEER TYPE     | STATE |  SINCE   |    INFO     |
+-----------------+-------------------+-------+----------+-------------+
| 2.2.2.2         | node-to-node mesh | up    | 11:01:07 | Established |
| 3.3.3.3         | node-to-node mesh | up    | 11:01:09 | Established |
+-----------------+-------------------+-------+----------+-------------+
```
on nodes1 (2.2.2.2):
```
sudo calicoctl node status
+-----------------+-------------------+-------+----------+-------------+
|  PEER ADDRESS   |     PEER TYPE     | STATE |  SINCE   |    INFO     |
+-----------------+-------------------+-------+----------+-------------+
| 1.1.1.1         | node-to-node mesh | up    | 11:01:07 | Established |
| 3.3.3.3        | node-to-node mesh | up    | 11:01:09 | Established |
+-----------------+-------------------+-------+----------+-------------+
```
on nodes2 (3.3.3.3):
```
sudo calicoctl node status
+-----------------+-------------------+-------+----------+-------------+
|  PEER ADDRESS   |     PEER TYPE     | STATE |  SINCE   |    INFO     |
+-----------------+-------------------+-------+----------+-------------+
| 1.1.1.1         | node-to-node mesh | up    | 11:01:07 | Established |
| 2.2.2.2         | node-to-node mesh | up    | 11:01:09 | Established |
+-----------------+-------------------+-------+----------+-------------+
```
the logs of the calico pods for node1 are :
Readiness probe failed: calico/node is not ready: BIRD is not ready: BGP not established with 3.3.3.3,1.1.1.12020-01-18 11:01:31.480 [INFO][160] health.go 156: Number of node(s) with BGP peering established = 0

Readiness probe failed: calico/node is not ready: BIRD is not ready: BGP not established with 1.1.1.1,3.3.3.32020-01-18 11:01:31.480 [INFO][160] health.go 156: Number of node(s) with BGP peering established = 1

and the same log for ndoe2 

in calico.yaml I have set the   IP_AUTODETECTION_METHOD to "interface=eth.*"
```
  - name: IP_AUTODETECTION_METHOD
              value: "interface=eth.*"            
```
also I have test the `value: value: can-reach=8.8.8.8`. but nothing changed.

I dont know what is my fault . tunnels cannot see each other 
and each node only can see its pods, not the other node's pod

also my pod network is 12.0.0.0/16 
but the pods and tunnels get the IP with the /32 netmask like: 12.0.66.128 with netmask 255.255.255.255

**Comment by ehsanrocket** [comment]
I found the problem
after pinging from master to nodes 
nodes can see each other and each other's pod. 
but the it is a problem itself
is there any thing I should do in calico.yaml ?

**Comment by tmjd** [comment]
What platform are your hosts running on? (AWS, GCE, Baremetal, Openstack) At one point I think there was an issue where it was necessary to ping between pods in certain setups but I thought that was fixed.

Here is the other issue I was thinking of https://github.com/projectcalico/calico/issues/1996, please read through that and see if that helps.

**Comment by ehsanrocket** [comment]
my platform is Baremetal
when the traffic goes to the nodes, nodes can communicate with it's pod and other nod's pod.
but I have deleted all the firewall rules from the edge firewall. (not the rules created by calico) that all ports and protocols are allowed.
I think Node's should be able to connect tot the Master to these ports: 179, 2379, 2380, 10250, 10251, 10252, 10256, 6443 
and Master should be able to connect to the Nodes to these ports: 179, 10250, 10256 
am I right? 
is there any other ports I should consider If I want to create firewall rules?

**Comment by tmjd** [comment]
Have you figured out your original problem of needing to ping from your master to your nodes before pod networking worked?

If you are trying to understand the ports that are required for Calico you should refer to https://docs.projectcalico.org/getting-started/kubernetes/requirements#network-requirements. Note this does not include the ports that kubernetes needs for communication.

**Comment by ehsanrocket** [comment]
before pod networking , I have ping my nodes from the master node and vice versa.
thank for networking requirements. I will check them.

**Comment by ehsanrocket** [comment]
@tmjd Our platrform is OpenStack 
I have said that it is BareMetal . sorry for wrong information.

with no firewall rules: 

ping From Node A tunl0  => Node B tunl0  => no ping 
at the same time 
ping From Node B tunl0 => Node A tunl0 => ping works at the both ends

so after adding a new node to cluster, for sending the traffic to the pods of the new node I should do the pings so that the nodes pod became reachable.

**Comment by tmjd** [comment]
I feel like this is probably a similar problem as the https://github.com/projectcalico/calico/issues/1996 one I linked above. Have you tried verifing that your OpenStack is really allowing IPIP traffic between hosts (before doing the pings)? Several years ago I was doing an OpenStack setup and IIRC, I actually had to do some configuration under the hood (on the routers themselves) to enable IPIP traffic in a cluster I was setting up.

The only way I can think of to kind of test that is use tcpdump to see IP-in-IP traffic that is destined for a remote host and then go use tcpdump on the remote host and verify the traffic is not arriving. I'm not sure how to manually generate IP-in-IP traffic to test it otherwise.
